tests/components/samsungtv/__init__.py

```diff
@@ -1,17 +1,72 @@
 """Tests for the samsungtv component."""
 from __future__ import annotations
 
+from datetime import timedelta
 from unittest.mock import Mock
 
 from async_upnp_client.client import UpnpAction, UpnpService
 
-from homeassistant.components.samsungtv.const import DOMAIN
+from homeassistant.components import ssdp
+from homeassistant.components.samsungtv.const import (
+    CONF_MODEL,
+    DOMAIN,
+    ENTRY_RELOAD_COOLDOWN,
+    METHOD_WEBSOCKET,
+)
+from homeassistant.components.ssdp import (
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_MANUFACTURER,
+    ATTR_UPNP_MODEL_NAME,
+    ATTR_UPNP_UDN,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_METHOD, CONF_NAME, CONF_PORT
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.typing import ConfigType
 from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as dt_util
 
-from tests.common import MockConfigEntry
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+MOCK_WS_ENTRY = {
+    CONF_HOST: "fake_host",
+    CONF_METHOD: METHOD_WEBSOCKET,
+    CONF_PORT: 8002,
+    CONF_MODEL: "any",
+    CONF_NAME: "any",
+}
+
+MOCK_SSDP_DATA_RENDERING_CONTROL_ST = ssdp.SsdpServiceInfo(
+    ssdp_usn="mock_usn",
+    ssdp_st="urn:schemas-upnp-org:service:RenderingControl:1",
+    ssdp_location="https://fake_host:12345/test",
+    upnp={
+        ATTR_UPNP_FRIENDLY_NAME: "[TV] fake_name",
+        ATTR_UPNP_MANUFACTURER: "Samsung fake_manufacturer",
+        ATTR_UPNP_MODEL_NAME: "fake_model",
+        ATTR_UPNP_UDN: "uuid:0d1cef00-00dc-1000-9c80-4844f7b172de",
+    },
+)
+MOCK_SSDP_DATA_MAIN_TV_AGENT_ST = ssdp.SsdpServiceInfo(
+    ssdp_usn="mock_usn",
+    ssdp_st="urn:samsung.com:service:MainTVAgent2:1",
+    ssdp_location="https://fake_host:12345/test",
+    upnp={
+        ATTR_UPNP_FRIENDLY_NAME: "[TV] fake_name",
+        ATTR_UPNP_MANUFACTURER: "Samsung fake_manufacturer",
+        ATTR_UPNP_MODEL_NAME: "fake_model",
+        ATTR_UPNP_UDN: "uuid:0d1cef00-00dc-1000-9c80-4844f7b172de",
+    },
+)
```

Comment:
MOCK_WS_ENTRY => MOCK_ENTRYDATA_WS

And can we move all three to const.py?

Reply:
Feel free to push to this. Family is here so I won't be able to look until right before bed